tests/src/common/TestUtils.java

```diff
@@ -287,87 +305,108 @@ public String toString() {
+    public static Pair<String, String> runCmd(int expectedExitCode, File dir, String... params) throws IOException {
         return runCmd(expectedExitCode, dir, logger, null, params).logs();
     }
 
-    public static Pair<String, String> runQuietly(int expectedExitCode, File dir, String... params) throws IllegalArgumentException, IOException {
+    public static Pair<String, String> runQuietly(int expectedExitCode, File dir, String... params) throws IOException {
         return runCmd(expectedExitCode, dir, null, null, params).logs();
     }
 
     /*
      * Run with no timeout.
      */
-    public static RunResult runCmd(int expectedExitCode, File dir, Logger logger,
-                                   Map<String, String> env, String... params) throws IllegalArgumentException, IOException {
+    public static RunResult runCmd(int expectedExitCode, File dir, Logger logger, Map<String, String> env, String... params) throws IOException {
         return runCmd(expectedExitCode, 0, dir, logger, env, params);
     }
 
     /**
      * Run a command in another process (exec())
      *
-     * @param expectedExitCode the exit code expected from the command when it exists
-     * @param timeoutMilli kill the underlying process after this amount of time (0 if no timeout)
-     * @param dir the working directory the command runs with
-     * @param logger object to manage logging message
-     * @param env TODO
-     * @param params parameters to pass on the command line to the spawnded command
+     * @param expectedExitCode
+     *            the exit code expected from the command when it exists
+     * @param timeoutMilli
+     *            kill the underlying process after this amount of time (0 if no
+     *            timeout)
+     * @param dir
+     *            the working directory the command runs with
+     * @param logger
+     *            object to manage logging message
+     * @param env
+     *            TODO
+     * @param params
+     *            parameters to pass on the command line to the spawnded command
      * @return
+     * @throws IOException
      */
-    public static RunResult runCmd(int expectedExitCode, int timeoutMilli, File dir, Logger logger,
-                                   Map<String, String> env, String... params) throws IllegalArgumentException, IOException {
+    public static RunResult runCmd(int expectedExitCode, int timeoutMilli, File dir, Logger logger, Map<String, String> env, String... params) throws IOException {
+        ProcessBuilder pb = new ProcessBuilder(params);
+        pb.directory(dir);
+        if (env != null)
+            pb.environment().putAll(env);
+        Process p = pb.start();
 
-        BasicLauncher bl = new BasicLauncher(true, true, logger);
-        bl.setCmd(params);
-        bl.setWorkingDir(dir);
-        bl.setEnv(env);
+        String stdout = inputStreamToString(p.getInputStream());
+        String stderr = inputStreamToString(p.getErrorStream());
```

Comment:
I'll merge this but for the record, the draining of the streams in this way may be prone to hiccups (basic launcher drained these streams in parallel using threads).